tests/safe_mode/data/db1.increment.sql

```diff
@@ -1,9 +1,11 @@
 use safe_mode_test;
 insert into t1 (uid, name) values (10003, 'Buenos Aires');
 alter table t1 add column age int;
+start transaction;
 insert into t1 (uid, name, age) values (10005, 'Buenos Aires', 200);
 insert into t2 (uid, name) values (20005, 'Aureliano José');
 insert into t1 (uid, name, age) values (10006, 'Buenos Aires', 200);
+commit;
```

Comment:
Unfortunately, this case does not cover the wrong situation. The case can be passed even on the master branch before #1329  merged. The reason is source1 enable gtid, you can add the case for source2 or disable gtid for source1.